ci/tests/csvpump/test.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash  
+
+set -eao pipefail
+
+function usage {
+    local progname=$1
+    cat <<EOF
+Usage:
+
+    $progname  tag [gwtag]
+
+    tag   -  The tag for the pump image, that needs to be tested.
+    gwtag -  (Optional) The  tag for the GW image, that needs to be tested against the pump.
+             If not given, the default version would be used, which is $DEFAULT_GW_TAG
+
+Brings up the given tyk gw, and tyk pump and checks if the csv pump is
+working properly.
+Requires docker compose.
+
+EOF
+    exit 1
+}
+
+function info {
+    local text=$1
+    echo -e "\033[0;36m $text \033[0m"
+}
+
+function warn {
+    local text=$1
+    echo -e "\033[0;31m $text \033[0m"
+}
+
+# The default gateway image that will be pulled and checked against in case
+# no explicit gateway image tag is given.
+DEFAULT_GW_TAG=v4.0.1
+
+[[ -z $1 ]] && usage "$0"
+export tag=$1
+export gwtag=$2
+
+if [[ -z $2 ]]; then
+    warn "No explicit gateway tag given - Will use $DEFAULT_GW_TAG"
+    gwtag=$DEFAULT_GW_TAG
+fi
+
+compose='docker-compose'
+# use the compose client plugin if v2
+[[ $(docker version --format='{{ .Client.Version }}') =~ 20.10 ]] && compose='docker compose'
+
+#create the tmp directory to hold pump data.
+if [[ -e /tmp/pump-data ]]; then
+    rm -rf /tmp/pump-data/*.csv
+else
+    mkdir /tmp/pump-data
+fi
+
+trap '$compose down' EXIT
+
+$compose up -d
+
+GWBASE="http://localhost:8080"
+
+
+curlf() {
+    curl --header 'content-type:application/json' -s --show-error "$@"
+}
```

Comment:
Lets add real http code failure control here:
```suggestion
curlf() {
  OUTPUT_FILE=$(mktemp)
  HTTP_CODE=$(curl --header 'Content-Type:application/json' --silent --output $OUTPUT_FILE --write-out "%{http_code}" "$@")
  if [[ ${HTTP_CODE} -lt 200 || ${HTTP_CODE} -gt 299 ]] ; then
    >&2 cat $OUTPUT_FILE
    return 22
  fi
  cat $OUTPUT_FILE
  rm $OUTPUT_FILE
}
```

Reply:
Since the user agent grep is the verification condition, it doesn't matter even if curl fails silently without an exit code. Unless the curl command resulted in a record, the data won't get exported, and the pump test will anyway fail.